src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -1786,6 +1789,9 @@ await PreviewBuildIntegratedProjectActionsAsync(buildIntegratedProject, actions,
                     if (msbuildProject != null)
                     {
                         msbuildProject.MSBuildNuGetProjectSystem.EndProcessing();
+
+                        // raise nuget batch end event
+                        msbuildProject.NotifyEndBatchProcessing();
```

Comment:
nit: this event should be before the project system end, it's easier to make sense of these events by thinking of which is inner vs outer. Here it's random.